src/apps/chifra/cmd/serve/exec/admin_pins.go

```diff
@@ -0,0 +1,62 @@
+package serveRoutes
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/cmd/pins"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/pinlib"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/pinlib/manifest"
+)
+
+var ErrMultipleModes = errors.New("only one of ?list or ?init is supported")
+var ErrNoMode = errors.New("you must choose one of ?list or ?init")
+
+// AdminPins runs pin-operating functions depending on query parameters
+func AdminPins(request *http.Request) ([]manifest.PinDescriptor, error) {
+	query := request.URL.Query()
+	mode := ""
+	all := false
+
+	for key := range query {
```

Comment:
I'm not sure this is what we want to do. As you know all of the 'command line' options code is programmatically generated from a .csv file. By putting this code here (as opposed to say centralizing it inside of the `pins` package, we've created a second (actually third) place where the auto-generation code will eventually have to look.

I know it seems a bit troublesome, but I cannot tell you have wonderful it is to have the auto-generation stuff. In the same way you can be confident that changes to your code don't get broken if you have good testing and that allows you to be ultra-flexible, being able to autogenerate code for all command-line options -- especially documentation -- allows for complete flexibility in changing command lines.

Upshot: can we somehow move this into either a common function that validates the command lines (even if this means refactoring the existing `validatePins` routine to take a string arrays, or, at the very least put this code into the package so it's all in one place (as opposed to in the `serve` subfolder?

Reply:
Good point. I will try to reuse the existing code we have for validating command line arguments